libraries/Microsoft.Bot.Builder.Dialogs/DialogContext.cs

```diff
@@ -256,16 +244,15 @@ private async Task EndActiveDialogAsync(DialogReason reason, CancellationToken c
             }
         }
 
-        private DialogTurnResult VerifyTurnResult(DialogTurnResult result)
-        {
-            result.HasActive = Stack.Count() > 0;
-            if (result.HasActive)
-            {
-                result.HasResult = false;
-                result.Result = null;
-            }
-
-            return result;
-        }
+        // private DialogTurnResult VerifyTurnResult(DialogTurnResult result)

```

Comment:
So this was simply because I don't trust the status that comes back from the dialog. I'm ok removing this for now, knowing that we can add it back end if it turns into an issue.